### PR TITLE
<GSS Closed Loop BZ: 2264900> Tests to verify restore a pvc from snapshot when the parent PVC is deleted

### PR DESCRIPTION
**Pull request by Akarsha-rai**

### CODE REVIEW

File: tests/functional/pv/pvc_snapshot/test_restore_snapshot_when_parent_pvc_deleted.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+)
+from ocs_ci.ocs.resources import pod
+from ocs_ci.helpers import helpers
+
+log = logging.getLogger(__name__)
+
+
+@green_squad
+@tier1
+@skipif_ocs_version("<4.16")
```
Comment (jilju) [inline tests/functional/pv/pvc_snapshot/test_restore_snapshot_when_parent_pvc_deleted.py:22]: Isn't the test case supported in lower versions ?

Comment (Akarsha-rai) [reply tests/functional/pv/pvc_snapshot/test_restore_snapshot_when_parent_pvc_deleted.py:22]: I see the [bug2264900](url) fixed in 4.16, so I believe it is not supported in lower version.

**Review by jilju — APPROVED** [review]

File: tests/functional/pv/pvc_snapshot/test_restore_snapshot_when_parent_pvc_deleted.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        for snap_obj in snap_objs:
+            access_modes = access_modes_dict[snap_obj.interface][
+                snap_obj.parent_volume_mode
+            ]
+            for access_mode in access_modes:
+                restore_obj = snapshot_restore_factory(
+                    snapshot_obj=snap_obj,
+                    volume_mode=snap_obj.parent_volume_mode,
+                    access_mode=access_mode,
+                    status="",
```
Comment (PrasadDesala) [inline tests/functional/pv/pvc_snapshot/test_restore_snapshot_when_parent_pvc_deleted.py:152]: any reason for not checking the restored PVC status for bound state here ?

File: tests/functional/pv/pvc_snapshot/test_restore_snapshot_when_parent_pvc_deleted.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+        restore_pvcs = []
+        for snap_obj in snap_objs:
+            access_modes = access_modes_dict[snap_obj.interface][
+                snap_obj.parent_volume_mode
+            ]
+            for access_mode in access_modes:
+                restore_obj = snapshot_restore_factory(
+                    snapshot_obj=snap_obj,
+                    volume_mode=snap_obj.parent_volume_mode,
+                    access_mode=access_mode,
```
Comment (PrasadDesala) [inline tests/functional/pv/pvc_snapshot/test_restore_snapshot_when_parent_pvc_deleted.py:151]: how are you making sure that we are not restoring the snapshot with the same access mode? To me, it looks like with the current approach, there will be a few restores with same access mode.

File: tests/functional/pv/pvc_snapshot/test_restore_snapshot_when_parent_pvc_deleted.py (hunk 2 above)
Comment (Akarsha-rai) [reply tests/functional/pv/pvc_snapshot/test_restore_snapshot_when_parent_pvc_deleted.py:152]: It takes more time to reach Bound state, to save time, Bound status is verified as next step

**Review by PrasadDesala — APPROVED** [review]

**Review by jilju — APPROVED** [review]

**Pull request merged**